pkg/store/memory_store.go

```diff
@@ -161,3 +179,83 @@ func (s *InMemoryStore) GetUpdates() []upstreamtypes.ChannelRelease {
 func (s *InMemoryStore) SetUpdates(updates []upstreamtypes.ChannelRelease) {
 	s.updates = updates
 }
+
+func (s *InMemoryStore) CreateInstanceReportEvent(event heartbeattypes.InstanceReportEvent) error {
+	existingSecret, err := s.clientset.CoreV1().Secrets(s.GetNamespace()).Get(context.TODO(), InstanceReportSecretName, metav1.GetOptions{})
+	if err != nil && !kuberneteserrors.IsNotFound(err) {
+		return errors.Wrap(err, "failed to get airgap instance report secret")
+	} else if kuberneteserrors.IsNotFound(err) {
+		instanceReport := &heartbeattypes.InstanceReport{
+			Events: []heartbeattypes.InstanceReportEvent{event},
+		}
+		data, err := instanceReport.Encode()
+		if err != nil {
+			return errors.Wrap(err, "failed to encode instance report")
+		}
+
+		uid, err := util.GetReplicatedDeploymentUID(s.clientset, s.GetNamespace())
+		if err != nil {
+			return errors.Wrap(err, "failed to get replicated deployment uid")
+		}
+
+		secret := &corev1.Secret{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "ConfigMap",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      InstanceReportSecretName,
+				Namespace: s.GetNamespace(),
+				OwnerReferences: []metav1.OwnerReference{
```

Comment:
might be worth adding a comment here to explain why owner references are needed here.